libs/ts-rest/core/src/lib/client.ts

```diff
@@ -67,6 +67,20 @@ export type ApiRouteResponse<T> =
       body: unknown;
     };
 
+export type ApiResponseForRoute<T extends AppRoute> =
+  | {
+      [K in keyof T['responses']]: {
+        status: K
+        body: ZodInferOrType<T['responses'][K]>
+      }
+    }[keyof T['responses']]
+  & {
+      status: Exclude<HTTPStatusCode, keyof T>
+      body: unknown
+    };
+
```

Comment:
I believe the `Intersection Type` makes more sense for response types and the current implementation.